### PR TITLE
New Resource: `azurerm_dev_center_gallery`

### PR DESCRIPTION
**Pull request by jiaweitao001**
* Generated by Pandora code gen tool.

* All acc tests passed.

```
=== RUN   TestAccDevCenterGallery_basic
=== PAUSE TestAccDevCenterGallery_basic
=== RUN   TestAccDevCenterGallery_requiresImport
=== PAUSE TestAccDevCenterGallery_requiresImport
=== RUN   TestAccDevCenterGallery_complete
=== PAUSE TestAccDevCenterGallery_complete
=== RUN   TestAccDevCenterGallery_update
=== PAUSE TestAccDevCenterGallery_update
=== CONT  TestAccDevCenterGallery_basic
=== CONT  TestAccDevCenterGallery_complete
=== CONT  TestAccDevCenterGallery_update
=== CONT  TestAccDevCenterGallery_requiresImport
--- PASS: TestAccDevCenterGallery_update (1137.43s)
--- PASS: TestAccDevCenterGallery_basic (1231.93s)
--- PASS: TestAccDevCenterGallery_complete (1424.79s)
--- PASS: TestAccDevCenterGallery_requiresImport (1471.54s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/devcenter     1488.768s
```

### CODE REVIEW

**Review by stephybun — COMMENTED** [review]
@jiaweitao001 - I'm unsure if this PR was opened so that we can see the output of the generated resource from Pandora, but just to clarify:

_pandora generated resources are added to the azurerm repo by an automatically generated PR, they are not manually opened PRs with the resource output copied into it_

Please close this PR, review will continue in https://github.com/hashicorp/pandora/pull/3271

File: internal/services/devcenter/dev_center_gallery_resource_gen.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+type DevCenterGalleryResource struct{}
+
+func (r DevCenterGalleryResource) ModelObject() interface{} {
+	return &DevCenterGalleryResourceSchema{}
+}
+
+type DevCenterGalleryResourceSchema struct {
+	DevCenterId       string `tfschema:"dev_center_id"`
+	GalleryResourceId string `tfschema:"gallery_resource_id"`
```
Comment (stephybun) [inline internal/services/devcenter/dev_center_gallery_resource_gen.go:27]: Like we did for the generated resource let's rename this to `shared_gallery_id`
```suggestion
	GalleryResourceId string `tfschema:"shared_gallery_id"`
```

File: internal/services/devcenter/dev_center_gallery_resource_gen.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	return "azurerm_dev_center_gallery"
+}
+func (r DevCenterGalleryResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"dev_center_id": {
+			ForceNew: true,
+			Required: true,
+			Type:     pluginsdk.TypeString,
+		},
+		"gallery_resource_id": {
```
Comment (stephybun) [inline internal/services/devcenter/dev_center_gallery_resource_gen.go:44]:
```suggestion
		"shared_gallery_id": {
```

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks @jiaweitao001, the tests look good! Just need to rename the `gallery_resource_id` property (don't forget to update it in the tests and docs) and then this should be good to go!

**Review by stephybun — APPROVED** [review]
Thanks @jiaweitao001 LGTM 👍

**Comment by tombuildsstuff** [comment]
@jiaweitao001 FYI the generated files shouldn't be added by hand - the automation regenerates these based [on the files within the Pandora repository](https://github.com/hashicorp/pandora/blob/main/config/resources/devcenter.hcl) - so if these aren't defined there [they'll be removed by the automation when it next runs, as is happening here](https://github.com/hashicorp/terraform-provider-azurerm/pull/24805) - as such we need to add the definition for this Resource, so this doesn't get removed.

**Comment by jiaweitao001** [comment]
> @jiaweitao001 FYI the generated files shouldn't be added by hand - the automation regenerates these based [on the files within the Pandora repository](https://github.com/hashicorp/pandora/blob/main/config/resources/devcenter.hcl) - so if these aren't defined there [they'll be removed by the automation when it next runs, as is happening here](https://github.com/hashicorp/terraform-provider-azurerm/pull/24805) - as such we need to add the definition for this Resource, so this doesn't get removed.

Hi @tombuildsstuff , thanks for the reminder. This resource was added manually because we had trouble on generating the code with Pandora code gen tool. I'm wondering if we add a definition for the resource in Pandora, is the auto gen tool gonna skip this resource or overwrite it?

**Comment by tombuildsstuff** [comment]
@jiaweitao001 forgot to circle back here after fixing this yesterday - apologies.

Reading the context in the other threads I ended up opening https://github.com/hashicorp/terraform-provider-azurerm/pull/24806 to rename these files so these are no longer generated - which'll fix this in a different manner.

**Pull request merged**